test/cppsuite/tests/search_near_01.cxx

```diff
@@ -170,116 +173,130 @@ class search_near_01 : public test_harness::test {
+             * the number of z key searches we have done this iteration.
+             */
+            if (srch_key == "z" || srch_key == "zz" || srch_key == "zzz")
+                ++z_key_searches;
+            tc->transaction.rollback();
+        }
+    }
+
     void
     read_operation(test_harness::thread_context *tc) override final
     {
         /* Make sure that thread statistics cursor is null before we open it. */
         testutil_assert(tc->stat_cursor.get() == nullptr);
-        logger::log_msg(
-          LOG_INFO, type_string(tc->type) + " thread {" + std::to_string(tc->id) + "} commencing.");
-        std::map<uint64_t, scoped_cursor> cursors;
-        tc->stat_cursor = tc->session.open_scoped_cursor(STATISTICS_URI);
-        std::string srch_key;
-        int64_t entries_stat, prefix_stat, prev_entries_stat, prev_prefix_stat, expected_entries,
-          buffer;
-        int cmpp;
+        /* This test will only work with one read thread. */
+        testutil_assert(tc->thread_count == 1);
+        test_harness::configuration *workload_config, *read_config;
+        std::vector<thread_context *> workers;
+        std::atomic<int64_t> z_key_searches;
+        int64_t entries_stat, expected_entries, prefix_stat, prev_entries_stat, prev_prefix_stat;
+        int num_threads;
 
-        cmpp = 0;
         prev_entries_stat = 0;
         prev_prefix_stat = 0;
+        num_threads = _config->get_int("search_near_threads");
+        tc->stat_cursor = tc->session.open_scoped_cursor(STATISTICS_URI);
+        workload_config = _config->get_subconfig(WORKLOAD_GENERATOR);
+        read_config = workload_config->get_subconfig(READ_CONFIG);
+        z_key_searches = 0;
+
+        logger::log_msg(LOG_INFO, type_string(tc->type) + " thread commencing.");
 
         /*
          * The number of expected entries is calculated to account for the maximum allowed entries
          * per search near function call. The key we search near can be different in length, which
          * will increase the number of entries search by a factor of 26.
          */
-        expected_entries =
-          tc->thread_count * keys_per_prefix * pow(ALPHABET.size(), PREFIX_KEY_LEN - srchkey_len);
-
-        /*
-         * Read at timestamp 10, so that no keys are visible to this transaction. This allows prefix
-         * search near to early exit out of it's prefix range when it's trying to search for a
-         * visible key in the tree.
-         */
-        tc->transaction.begin("read_timestamp=" + tc->tsm->decimal_to_hex(10));
+        expected_entries = keys_per_prefix * pow(ALPHABET.size(), PREFIX_KEY_LEN - srchkey_len);
         while (tc->running()) {
+            runtime_monitor::get_stat(
+              tc->stat_cursor, WT_STAT_CONN_CURSOR_NEXT_SKIP_LT_100, &prev_entries_stat);
+            runtime_monitor::get_stat(tc->stat_cursor,
+              WT_STAT_CONN_CURSOR_SEARCH_NEAR_PREFIX_FAST_PATHS, &prev_prefix_stat);
```

Comment:
Minor: we don't have to get stats twice every iteration. Only the first one doesn't have any prev data and that's the only case when we need to get stats at the beginning of the iteration.

Reply:
Ahh good point! I actually want to be as confident as we can at this point, in the case that there are search nears performed from WT internally that could potentially increase this stat.

I am fine for both cases here, let me know if we should change it instead.

Reply:
As I said, it's a minor, we can leave it as is. Not a big deal in terms of performance for the test.